source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -105,6 +132,77 @@ Extractor::extractValue(Http::StreamFilterCallbacks &callbacks,
   return "";
 }
 
+// Match a regex against the input value and replace the matched subgroup with the replacement_text_ value
+// writes the result to replaced_value_ and returns a absl::string_view to it
+absl::string_view
+Extractor::replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                  absl::string_view value) const {
+  std::match_results<absl::string_view::const_iterator> regex_result;
+
+  // if there are no matches, return an empty string
+  if (!std::regex_search(value.begin(), value.end(), regex_result, extract_regex_)) {
+    ENVOY_STREAM_LOG(debug, "replaceValue: extractor regex did not match input", callbacks);
+    return "";
+  }
+
+  // if the subgroup specified is greater than the number of subgroups in the regex, return an empty string
+  if (group_ >= regex_result.size()) {
+    ASSERT("no such group in the regex");
+    ENVOY_STREAM_LOG(debug, "replaceValue: invalid group specified for regex", callbacks);
+    return "";
+  }
+
+  // if the regex doesn't match the entire input value, return an empty string
+  if (regex_result[0].length() != long(value.length())) {
+    ENVOY_STREAM_LOG(debug, "replaceValue: Regex did not match entire input value. This is not allowed when replace_all is false", callbacks);
+    return "";
+  }
+
+  // Create a new string with the input value
+  std::string replaced(value.begin(), value.end());
+
+  // Replace the specified subgroup with the replacement_text_ value
+  const auto &sub_match = regex_result[group_];
+  auto start_pos = sub_match.first - value.begin();
+  auto length = sub_match.length();
+  replaced.replace(start_pos, length, replacement_text_);
+
+  // Store the replaced string in replaced_value_ and return an absl::string_view to it
+  replaced_value_ = std::move(replaced);
+  return absl::string_view(replaced_value_);
```

Comment:
> What would happen if the replacement text is longer than the regex
  - This is a very good point. I will add a test against this. I don't know how `.replace` will behave + I do expect we would need an additional allocation even if the results are correct. I'm going to think about coming up w/ a more robust way to handle this

Reply:
Updated here to prevent strings from being reallocated: 55ddd1bfc4d9976bc953cab2dd0fcc874edd9405

```cpp
  // Create a new string with the maximum possible length after replacement
  auto max_possible_length = value.length() + replacement_text_.value().length();
  std::string replaced;
  replaced.reserve(max_possible_length);

  auto subgroup_start = regex_result[group_].first;
  auto subgroup_end = regex_result[group_].second;

  // Copy the initial part of the string until the match
  replaced.assign(value.begin(), subgroup_start);

  // Append the replacement text
  replaced += replacement_text_.value();

  // Append the remaining part of the string after the match
  replaced.append(subgroup_end, value.end());
```